Updated for game version 0.8.1

nickmcski opened the pull request:
Updated the harmony hooks to work with game version 0.8.1. I expect this will break compatibility with previous game versions.

nickmcski committed:
Updated for game version 0.8.1

nickmcski committed:
Bump Harmony, remove dependancy on splat

grasmanek94 commented:
![image](https://user-images.githubusercontent.com/595329/103803441-dbca3500-5050-11eb-974c-3783c0d750dd.png)
I think there's still something wrong with the food / item use amount.
Apart from that, nice work in updating it, thanks. When I find time I can update it, or maybe you can also update the PR to fix this.

grasmanek94 commented:
`public void OnVisit(NPCBase npc, out bool gotFood)` looks like the correct place, unfortunately it's one big glob of code, I'll try to come up with something.

grasmanek94 commented on StockpileStatisticsHooks.cs:
```diff
@@ -56,21 +56,21 @@ static void Postfix(Stockpile __instance, bool __result, ushort type, int amount
 		public static bool isFood;
 
-		static void Prefix(Stockpile __instance, ref float currentFood, float desiredFoodAddition)
+		static void Prefix(ref NPCBase npc, ref bool gotFood)
```
`npc` shouldn't be `ref` here, remove it.
gotFood.. I think we can make it `out`?

nickmcski replied:
I wasn't able to change gotFood to an out, it gave a compiler error as we aren't setting that value in our method. 

Would it be better to leave it as a ref, remove it, or is there a better way to get it as a out?

grasmanek94 commented on StockpileStatisticsHooks.cs:
```diff
@@ -56,21 +56,21 @@ static void Postfix(Stockpile __instance, bool __result, ushort type, int amount
 		}
 
-		static void Postfix(Stockpile __instance, bool __result, ref float currentFood, float desiredFoodAddition)
+		static void Postfix(ref NPCBase npc, ref bool gotFood)
```
Remove `ref` from `npc`, `gotFood` = `out`?

grasmanek94 commented on modInfo.json:
```diff
@@ -1,23 +1,28 @@
+    ],
 		"dllpath": "grasmanek94.Statistics.dll"
 	}
 ]
```
If you want:
could you please add a contributors.txt and add yourself in the format that you want?

grasmanek94 reviewed (changes requested):
a few small changes are also needed

grasmanek94 commented:
I was looking for the bug, and although I can't seem to find it at the moment, I did notice that we could add consumers for food with something like:

```
	[HarmonyPatch(typeof(ColonyShopVisitTracker))]
	[HarmonyPatch("OnVisit")]
	public class ColonyShopVisitTrackerHookOnVisit
    {
		public  static Stockpile inFunctionStockpile = null;
		public static bool isFood;
		public static NPCBase npc = null;

		static void Prefix(NPCBase arg_npc, bool gotFood)
		{
			// Log.WriteWarning("StockpileHookTryRemoveFood::Prefix");
			inFunctionStockpile = arg_npc.Colony.Stockpile;
			isFood = true;
			npc = arg_npc;
		}

		static void Postfix(NPCBase arg_npc, bool gotFood)
		{
			// Log.WriteWarning("StockpileHookTryRemoveFood::Postfix");
			inFunctionStockpile = null;
			isFood = false;
			npc = null;
		}
	}
// (...)
			if (ColonyShopVisitTrackerHookOnVisit.isFood)
			{
				AddConsumerRemoveProducer(itemStats, ColonyShopVisitTrackerHookOnVisit.npc, amount);

				itemStats.UseAsFood(amount);
			}
```

nickmcski committed:
Fix food usage calcuation error.
Code cleanup

grasmanek94 commented:
I see WIP? Lemme know when to review :) @nickmcski

nickmcski committed:
Fixed missing hook, add support for Foresters

nickmcski committed:
Updated readme and contributers

nickmcski commented:
> I see WIP? Lemme know when to review :) @nickmcski

I've moved those WIP changes into a separate branch, I just kept running into issues and realized it likely requires a larger change to how we are storing data. I'll open a separate discussion for that. 

I think I've addressed all your comments and changes so I would be happy if you could take another look!

grasmanek94 commented:
We do still have a bug where some food isn't counted.. like foofoo :/ but hey, at least most of it works.
